### DISCUSSION TITLE
Succsess: Adafruit Feather M0 Express + Ethernet FeatherWing

### DISCUSSION THREAD

**Discussion by NickLplus** [JAndrassy/ArduinoOTA#77] [Show and tell]
Basic OTE example works, with serial port disconnected it was necessary to modify
`while (!Serial );`
I did it this way:
```
bool SerialAvailable = false;
...
void setup() {
...
while (!Serial && (millis() < 4000));
if (Serial) SerialAvailable = true;

if (SerialAvailable) Serial.println("Initialize Ethernet with DHCP:");
```
inserting
if (SerialAvailable)
before each (block of) prints to serial.

If there is a lot of debug prints, a function may be more handy.

**Comment by JAndrassy** [comment]
don't you think it is off-topic here?

you can leave Serial.prints as they are. if USB is not connected they just don't do anything.

**Reply by NickLplus** [reply]
Sorry, I thought "Show and tell" is the right topic to let you know your library works with this hardware combination. I have zero github experience.

Serial was preventing the execution of loop, the sketch was just hanging with my Feather M0 Express. It didn't hang with HUZZAH.

Many thanks for the library and please excuse me if I did something wrong.

**Reply by JAndrassy** [reply]
sorry. thank you for reporting your succesfull use of the library with the Adafruit board and shield.
only the part about Serial is off-topic and I focused on it. sorry again.

the `while (!Serial);` is a normal thing. you need it to wait for Serial Monitor to connect, to catch debug print in setup(), and it should be removed if the sketch should run without USB connectiom
